snuba/utils/streams/kafka_consumer_with_commit_log.py

```diff
@@ -33,16 +37,21 @@ def __commit_message_delivery_callback(
         if error is not None:
             raise Exception(error.str())
 
-    def commit_offsets(self) -> Mapping[Partition, int]:
+    def commit_offsets(self) -> Mapping[Partition, Offset]:
         offsets = super().commit_offsets()
 
         for partition, offset in offsets.items():
-            commit = Commit(self.__group_id, partition, offset)
+            commit = Commit(self.__group_id, partition, offset.kafka_offset)
             payload = commit_codec.encode(commit)
             self.__producer.produce(
                 self.__commit_log_topic.name,
                 key=payload.key,
                 value=payload.value,
+                headers={
+                    "orig_message_ts": datetime.strftime(
+                        offset.timestamp, PAYLOAD_DATETIME_FORMAT
+                    ),
+                },
```

Comment:
What is the reason to use headers for this? 
Headers should be used for message metadata (basically stuff that would not break your application if they disappeared), or observability/routing information. 
If it is done in order to avoid having to change the consumers, I think we should do that investment and have a clean interface.
Is it about kafka compaction of this topic? 

Also doing this breaks the idea that we abstract the logic to generate payloads through a codec, thus increasing the coupling between the producing and consuming code.
If we need to store them as header, I think you should still add the timestamp to the Commit object and change the codec to extract headers

Reply:
Apart from putting it into the headers the other options would be to put it in the key or the body
- Key: I don't think we should put it in the key as we lose the benefits of log compaction.
- Body: The payload is relatively simple today (it's just the commit offset) and I was reluctant to make it more complex. If we put it the payload we probably want to convert that to json format. Unlike the header solution, he post process forwarder would then need to be aware of this and deserialise the json. To avoid this I have a slight preference to put the timestamp in the headers but this is not a strong opinion and I think there are good arguments for both ways.

I didn't update the commit object and codec because I didn't want arroyo to need to be aware of this but I can make that change.

Reply:
re: commit object. I would definitively do that. 

re: body. Mh, ok, I think eventually we will have to change the body structure anyway. It is ok if we do not do it right away considering the impact on consumers.